File: app/src/main/java/com/t895/freebie/MediaPlayerHelper.java
package com.t895.freebie;

import android.media.MediaPlayer;

import com.t895.freebie.models.Song;

public class MediaPlayerHelper
{
  public static final String TAG = "MediaPlayerService";

  public static MediaPlayer mediaPlayer;
  public static Song currentlyPlayingSong;

  public MediaPlayerHelper(UIManager manager)
  {
    if (mediaPlayer == null)
      mediaPlayer = new MediaPlayer();

    mediaPlayer.setOnCompletionListener(mp -> manager.songFinished());
    mediaPlayer.setOnPreparedListener(mediaPlayer -> manager.setActiveSong());
  }

  public static void setCurrentlyPlayingSong(Song song)
  {
    currentlyPlayingSong = song;
  }

  public static Song getCurrentlyPlayingSong()
  {
    return currentlyPlayingSong;
  }
}

File: app/src/main/java/com/t895/freebie/MediaPlayerHelper.kt
package com.t895.freebie

import androidx.media3.common.MediaItem
import androidx.media3.session.MediaController
import com.t895.freebie.models.Song

class MediaPlayerHelper
{
    private val TAG = "MediaPlayerHelper"

    companion object
    {
        var player: MediaController? = null
        private var currentlyPlayingSong: Song? = null

        fun playSong(song: Song)
        {
            player!!.removeMediaItems(0, 1)
            currentlyPlayingSong = song
            val mediaItem = MediaItem.Builder()
                .setUri(currentlyPlayingSong!!.uri.substring(5))
                .setMediaId(currentlyPlayingSong!!.title)
                .setTag(currentlyPlayingSong!!.title)
                .build()
            player!!.addMediaItem(mediaItem)
            player!!.play()
        }
    }

}
